src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ManagedToNativeVTableMethodGenerator.cs

```diff
@@ -75,34 +75,17 @@ internal sealed class ManagedToNativeVTableMethodGenerator
             }
 
             _context = new ManagedToNativeStubCodeContext(targetFramework, targetFrameworkVersion, ReturnIdentifier, ReturnIdentifier);
-            _marshallers = new BoundGenerators(argTypes, CreateGenerator);
+            _marshallers = new BoundGenerators(argTypes, generatorFactory, _context, new Forwarder());
```

Comment:
I do like this more, but I'm not sure why having a property is better than having an out on a factory method. Is there a reason you prefer the `new` allocation rather a `BoundGenerators.Create()`? The errors could be an `out` and be clear that they exist.

Reply:
I'll take another shot at trying to move this into a good factory design.

Reply:
It is merely a question about design. If you have a strong desire to avoid a `static` factory method, then I'm okay with that. Based on my rudimentary understanding here it seems the factory pattern is most appropriate, but I can be convinced.

Reply:
I don't have a particularly strong desire against moving this to a factory method, and I think it might end up being slightly cleaner. I just defaulted to keeping it as a constructor since that was an easier change.